public/javascripts/languageFilter.js

```diff
@@ -0,0 +1,144 @@
+var languageFilterComponent = (function() {
+    var addFilter = function(songs) {
+        m.request({
+                method: 'POST',
+                url: '/filter',
+                data: {
+                    langShown: langShown(),
+                    langFilter: langFilter()
+                }
+            })
+            .then(function(res) {
+                console.log(res.songs)
+                songs(res.songs)
+            })
+    }
+    var langShown = m.prop('English')
+    var langFilter = m.prop([])
+
+    var languageFilter = {
+        view: function(ctrl, args) {
+            return m('div', [
+                m('.btn-group', [
+                    m('button.btn.btn-default.dropdown-toggle[type=button]', {
+                        'data-toggle': "dropdown",
+                        'aria-haspopup': "true",
+                        'aria-expanded': "false"
+                    }, 'Language Filter: ' + langFilter().reduce((prev, curr, i) => {
+                        return (i === 0 ? prev + curr : prev + ', ' + curr)
+                    }, ''), [
+                        m('span.caret')
+                    ]),
+                    m('ul.dropdown-menu', [
+                        m('li', [
+                            m('label', [
+                                m('input#English[type=checkbox]', {
```

Comment:
Actually, change all this language specific code. Create an array of languages and generate the checkboxes from that.